moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -1949,6 +1949,12 @@ double moveit::core::RobotState::computeCartesianPath(const JointModelGroup* gro
     last_valid_percentage = percentage;
   }
 
+  // Check to see if the two first points in the trajectory are identical.
+  if (traj.size() > 1 && (*(*(traj[0])).position_) == (*(*(traj[1])).position_))
```

Comment:
The warning is more general than the test. Wouldn't make sense to test all waypoints?

Reply:
I could definitely check each waypoint. I was going back and forth on whether to do that or not.

I'll follow whatever you two decide is the correct way to solve this problem.

Reply:
Lets move this check into a ``validateCartesianInputWaypoints()`` function and have it check all points for repeats, sure. Be sure to add a comment explaining why this is important (it breaks the time parameterizer later)

Reply:
@davetcoleman I'll get this in ASAP!

Reply:
Just to clarify, you want a `validateCartesianInputWaypoints()` function now that gets called in `computeCartesianPath()`?

It would be a void function that tests the path and prints the warning? 

I want to verify I provide what you are asking for and nothing is lost in translation/I missed something in the large code base.

Reply:
I think the best thing to do would be to add a `dedupRobotTrajectory` method that takes in a `traj` such as the one generated by `computeCartesianPath` and removes duplicate waypoints. I would prefer if the helper function was called by `computeCartesianPath` before it returs

Reply:
Also, this syntax is completely unreadable